miner/worker.go

```diff
@@ -867,40 +862,44 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 
 	// Play our part in generating the random beacon.
 	if w.isRunning() && random.IsRunning() {
-		if randomSeed == nil {
-			account := accounts.Account{Address: w.coinbase}
-			wallet, err := w.eth.AccountManager().Find(account)
-			if err == nil {
-				// TODO: Use SignData instead
-				randomSeed, err = wallet.SignHash(account, common.BytesToHash(randomSeedString).Bytes())
-			}
-			if err != nil {
-				log.Error("Unable to create random seed", "err", err)
-				return
-			}
+		istanbul, ok := w.engine.(consensus.Istanbul)
+		if !ok {
+			log.Crit("Istanbul consensus engine must be in use for the randomness beacon")
+		}
+
+		lastCommitment, err := random.GetLastCommitment(w.coinbase, w.current.header, w.current.state)
+		if err != nil {
+			log.Error("Failed to get last commitment", "err", err)
+			return
+		}
+
+		lastRandomnessParentHash := rawdb.ReadRandomCommitmentCache(w.db, lastCommitment)
+		if (lastRandomnessParentHash == common.Hash{}) {
+			log.Error("Failed to get last randomness cache entry", "err", err)
```

Comment:
Coupled with when we write to the cache, this change makes it impossible to use a non-validator chaindata as a backup (I suspect that a replica node might save to the randomness cache, but I'm not positive yet).

Reply:
Ya, the reverse iteration logic that I am going to add will address the case of a validator using chaindata created from another node.